design/ceph/object/ceph-sse-s3.md

```diff
@@ -0,0 +1,44 @@
+---
+# AWS server side encryption SSE-S3 support for RGW
+target-version: release-1.10
+---
+
+# Feature Name
+AWS server side encryption SSE-S3 support for RGW
+
+## Summary
+The S3 protocol over three different types of [server side encryption](https://docs.aws.amazon.com/AmazonS3/latest/userguide/serv-side-encryption.html): SSE-C, SSE-KMS and SSE-S3. For the last two RGW server need to configure with external services such as `vault`. Currently Rook can configure RGW with `SSE-KMS` options, so that S3 request with `sse:kms` header will be handled properly. Recently the support for handling the `sse:s3` added to RGW with another set of options, so Rook needs to provide option to configure RGW with those options.
+
+### Goals
+Configure RGW with `SSE-S3` options, so that RGW can handle request with `sse:s3` headers.
+ 
+### Non-Goals
+Both `SSE-KMS` and `SSE-S3` will be configured with same set of values such as `vault-server`, `vault-secret-engine` etc
```

Comment:
This is a detail and not a non-goal. Further, how will we handle forward compatibility if these do need to be configured separately in the future.